packages/@react-spectrum/dropzone/docs/DropZone.mdx

```diff
@@ -0,0 +1,338 @@
+
+  return (
+    <>
+      <DraggableImage />
+      <DropZone
+        isFilled={isFilled}
+        maxWidth="size-3000"
+        height="size-2400"
+        UNSAFE_className={isFilled && 'is-filled'}
+        getDropOperation={(types) =>  (types.has('image/png') || types.has('image/jpeg')) ? 'copy' : 'cancel'}
+        onDrop={async (e) => {
+          e.items.find(async (item) => {
+            if (item.kind === 'file') {
+              if (item.type === 'image/jpeg' || item.type === 'image/png') {
+                setFilledSrc(URL.createObjectURL(await item.getFile()));
+                setIsFilled(true);
+              }
+            } else if (item.kind === 'text') {
+              setFilledSrc(await item.getText('image/jpeg'));
+              setIsFilled(true);
+            }
+          });
+        }}>
+        <IllustratedMessage>
+            <Upload />
+            <Heading>
+              <Text slot="label">
+                Drag and drop photos
+              </Text>
+            </Heading>
+        </IllustratedMessage>
+        {isFilled && <img className={'images'} alt="" src={filledSrc} />}
+      </DropZone>
+    </>
+  );
+}
+```
+
+<details>
+  <summary style={{fontWeight: 'bold'}}><ChevronRight size="S" /> Show CSS</summary>
+
+```css
+.images{
+  position: absolute;
+  top: 0px;
+  left: 0px;
+  width: 100%;
+  height: 100%;
+  object-fit: cover;
+  border-radius: var(--spectrum-alias-border-radius-small);
+}
+
+.is-filled {
+  border-style: solid;
+}
+
+```
+</details>
+
+The `DraggableImage` component used above is defined below. See [useDrag](../react-aria/useDrag.html) for more details and documentation.
+
+<details>
+  <summary style={{fontWeight: 'bold'}}><ChevronRight size="S" /> Show code</summary>
+
+```tsx example render=false export=true
+function DraggableImage() {
+  let {dragProps, isDragging} = useDrag({
+    getItems() {
+      return [
+        {
+          'image/jpeg': 'https://images.unsplash.com/photo-1528722828814-77b9b83aafb2?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8MTF8fHNwYWNlfGVufDB8fDB8fHww&auto=format&fit=crop&w=900&q=60'
+        }
+      ];
+    }
+  });
+
+  return (
+    <div
+      {...dragProps}
+      role="button"
+      tabIndex={0}
+      className={`draggable ${isDragging ? 'dragging' : ''}`} >
+      <img
+        width="150px"
+        height="100px"
+        alt="Cosmic Sky"
+        src="https://images.unsplash.com/photo-1528722828814-77b9b83aafb2?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8MTF8fHNwYWNlfGVufDB8fDB8fHww&auto=format&fit=crop&w=900&q=60"/>
+    </div>
+  );
+}
+```
+</details>
+
+### Replace message
+
+When a drop zone is in a filled state and has an object dragged over it, a message will appear in front of the drop zone. By default, this message will say "Drop file to replace". However, users can choose to customize this message through the `replaceMessage` prop. This message should describe the interaction that will occur when the object is dropped. 
```

Comment:
Might be good to mention that the message should be internationalized like we do on other component docs when discussing labeling.